tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,103 @@
+# Weight Clustering
```

Comment:
I'd update this based on https://www.tensorflow.org/model_optimization/guide/quantization/training
as a reference point.